### PR TITLE
Issue 23435 set layout to a page variant version

### LINKED ISSUE #23435: Set Layout to a Page Variant version
### Is your feature request related to a problem? Please describe.
We need to change the layout for a specific Page Variant version if the session storage varaintName item is set.

### Describe the solution you'd like
When the variantName session Storage item is set we need to send it as a query param to the end point that we use to create a new layout to a specific page

Also we need to support this new query param in the end point.

Also If the page not have any version on this variant we need to create it.

### Acceptance criteria
- Set a layout to a specific Page Variant Version

### PR DESCRIPTION
**Pull request by freddyucv**
### Proposed Changes
We need to change the Page's Layout just for a specific Variant, the changes need was:

* If the variantName sessionStorage item exists then sent it from the UI

https://github.com/dotCMS/core/pull/23577/files#diff-929579ea5524777449b560e6c0f9b6866b6c8dcefb51f9f41d0c21f549af4117R33

* If we want to save a layout to a Page into a specific Variant and this Page does not have any version in this Variant then we need to create this version:

https://github.com/dotCMS/core/pull/23577/files#diff-c74cebd0e6210e5d55c1e765543514a52f84f3f312277ddd3c80d21680fb545cR284

* If the Template is used by any other Page then we need to create a new one.

https://github.com/dotCMS/core/pull/23577/files#diff-b8bffe975249ff6e85fb62920437d75cf2ae572e1249729792a3fcdc9f0538edR420

* Right now when we change the Template for a Page automatically dotCMS change the Template for all the Page's Version in the different Languages, we are keeping this behavior but inside a Variant 

https://github.com/dotCMS/core/pull/23577/files#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82R5868

### CODE REVIEW

File: core-web/libs/data-access/src/lib/dot-page-layout/dot-page-layout.service.spec.ts [hunk 1]
```diff
@@ -5,8 +5,9 @@ import { DotPageLayoutService } from './dot-page-layout.service';
 import { HttpTestingController, HttpClientTestingModule } from '@angular/common/http/testing';
 import { CoreWebService } from '@dotcms/dotcms-js';
 import { CoreWebServiceMock, mockDotLayout } from '@dotcms/utils-testing';
+import { DotSessionStorageService } from '@dotcms/data-access';
 
-describe('DotPageLayoutService', () => {
+fdescribe('DotPageLayoutService', () => {
```
Comment (victoralfaro-dotcms) [inline core-web/libs/data-access/src/lib/dot-page-layout/dot-page-layout.service.spec.ts:10]: Typo?

**Review by victoralfaro-dotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java [hunk 2]
```diff
@@ -1787,4 +1792,173 @@ public void findDeletedContentletByIdentifierAnyLanguage()
...
         assertNull(contentletByIdentifierAnyLanguage);
     }
+
+    /**
+     * Method to test: {@link ESContentletAPIImpl#checkin(Contentlet, User, boolean)}
+     * When:
+     * - Create two {@link Language}: A and B
+     * - Create two {@link Template}: A and B
+     * - Create a {@link HTMLPageAsset} in the first language.
+     * - Create a new version of the same {@link HTMLPageAsset} but with different {@link Language}
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java:1802]: It's missing saying info about the page and template

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java [hunk 3]
```diff
@@ -407,4 +408,15 @@ List<Template> findAllVersions(final Identifier identifier, final User user, fin
 	 */
 	Template getTemplateByFolder(final Folder folder, final Host host, final User user, final boolean showLive) throws DotSecurityException, DotDataException;
 
+	/**
+	 * Return the List of Page's Inode tha used the {@link Template}
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java:412]: pages inode or actual pages?

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateFactoryImpl.java [hunk 4]
```diff
@@ -727,10 +729,38 @@ public Template getTemplateByFolder(final Host site, final Folder folder, final
...
+						.build()
+				)
+				.collect(Collectors.toList());
+	}
+
 	/**
 	 * Determine if the folder lives under /application/templates
 	 * @param folder to check
-	 * @return true if the folder lives under /application/templates
+	 * @return true if the folder lives under /application/templatesu
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateFactoryImpl.java:763]: typo?

File: dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java]: done https://github.com/dotCMS/core/pull/23577/commits/c88aeb826bfdbcbcf5fb893b31972ea4392000f7#diff-093de07a53a725b587963fd60c024f52e089c4f78e7ea88ba8e3a3a0cdaecab9R1801

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java]: done https://github.com/dotCMS/core/pull/23577/commits/c88aeb826bfdbcbcf5fb893b31972ea4392000f7#diff-b8bffe975249ff6e85fb62920437d75cf2ae572e1249729792a3fcdc9f0538edR412

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateFactoryImpl.java (hunk 4 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateFactoryImpl.java]: done https://github.com/dotCMS/core/pull/23577/commits/c88aeb826bfdbcbcf5fb893b31972ea4392000f7#diff-30db473047bf01b113d2ea4e19f591cd5d08cc41516f01e5bb1ec5be9131dfd7R763

**Review by dsilvam — APPROVED** [review]

**Pull request merged**